realm/realm-jni/build.gradle

```diff
@@ -248,6 +248,109 @@ task deployCore(group: 'build setup', description: 'Deploy the latest version of
     }
 }
 
+task compileCoreLibForHost(group: 'build setup', description: 'Compile the core library for host from source code') {
+    doLast {
+        if (!coreSourcePath) {
+            throw new GradleException("The Realm Core library is needed to build support for Robolectric. coreSourcePath, /usr/local/lib/librealm.a or the environment REALM_CORE_HOME is needed.")
+        }
+        exec {
+            workingDir = coreSourcePath
+            environment 'REALM_ENABLE_ENCRYPTION', 'yes'
+            commandLine = [
+                    "bash",
+                    "build.sh",
+                    "config"
+            ]
+        }
+        exec {
+            workingDir = coreSourcePath
+            commandLine = [
+                    "bash",
+                    "build.sh",
+                    "build"
+            ]
+        }
+
+        copy {
+            from "$coreSourcePath/src/realm/librealm.a"
+            into "${projectDir}/../realm-library/src"
+        }
+    }
+}
+
+task prepareCoreLibForHost(group: 'build setup', description: 'Prepare the core library for host') {
+    def staticLibHome = System.env.REALM_CORE_HOME ? System.env.REALM_CORE_HOME : '/usr/local'
+
+    doFirst {
+        if (file(staticLibHome).exists() == false) {
+            if (!coreSourcePath) {
+                throw new GradleException("The Realm Core library is needed to build support for Robolectric. coreSourcePath, /usr/local/lib/librealm.a or the environment REALM_CORE_HOME is needed.")
+            }
+        }
+    }
+
+    doLast {
+        copy {
+            from "$staticLibHome/lib/librealm.a"
+            into "${projectDir}/../realm-library/src"
+        }
+    }
+}
+
+task assembleRobolectric(group: 'build', description: "Build the Robolectric shared library for Darwin") {
+    dependsOn coreSourcePath ? compileCoreLibForHost : prepareCoreLibForHost
+
+    def staticLibHome = System.env.REALM_CORE_HOME ? System.env.REALM_CORE_HOME : '/usr/local'
+    def staticLibInclude = coreSourcePath ? "-I$coreSourcePath/src/" : "-I$staticLibHome/include/realm/"
+
+    doLast {
+        exec {
+            environment CC: "${clang ? 'clang' : 'gcc'}"
+            environment STRIP: "/usr/bin/strip -o librealm-jni-darwin-stripped.dylib"
+            commandLine = [
+                    'make',
+                    "-j${Runtime.getRuntime().availableProcessors() * 2}",
+                    "-l${Runtime.getRuntime().availableProcessors()}",
+                    '-C', "${projectDir}/src",
+                    "CC_IS=${clang ? 'clang' : 'gcc'}",
+                    "REALM_CFLAGS_COMMON=-Wno-variadic-macros -DREALM_HAVE_CONFIG -DPIC $staticLibInclude -I/usr/local/include/ -I/System/Library/Frameworks/JavaVM.framework/Headers/",
```

Comment:
`$staticLibInclude` is the core header directory indeed. I think `staticLibHome` should be `realmCoreHome` and `staticLibInclude` should be `realmCoureInclude` to prevent confusion.